enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,309 @@
+
+* `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
+* `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
+
+Future types may be:
+* `events.kubernetes` - Kubernetes events
+* `audit.container-engine` - Container engine audit events
+
+There are no assumptions regarding whether or not an endpoint is deployed on or off cluster.  Endpoints off-cluster may require adminstrators to perform additional actions in order for logs to be forwarded (e.g. secret creation, opening port) 
+Following is the list of supported endpoint types for this proposal:
+
+* Elasticsearch
+* Fluent secure-forward
+
+### User Stories
+
+#### As an OKD admin, I want to collect logs and send them to my own aggregation solution, for example Splunk
+
+This is a typical example of organizations that desires to re-use their existing enterprise log solution.  We will succeeded if we are able to write logs to their logging service.
+
+#### As an OKD admin, I want to aggregate application logs on-cluster and infra logs off-cluster
+
+This is an example of an OKD cluster hosting solution where several organizations are each provided with a dedicated cluster.  The organization requires access to application container logs but the host requires access to the infra structure logs.
+
+### Implementation Details
+
+#### Assumptions
+
+* A pipeline shall support at least one target endpoint. 
+* A pipeline may support multiple target endpoints.
+* Endpoints that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
+* Secrets shall have keys of: cert,key,cacert which point to the respective certificates for which they represent
+
+#### Security
+* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: cert, key, cacert
+* Endpoints defined without a secret will use unsecure communication
+
+#### Cluster Logging With Logstore, Collector, and LogForwarding
+* Undeclared source messages are routed to logstore
+* Declared source messages are routed to the defined service
+#### Cluster Logging With Pipelines Only
+* Undeclared source messages are ignored
+* Declared source messages are routed to the defined service
+
+
+### Risks and Mitigations
+
+## Design Details
+
+The `cluster-logging-operator` will use the forwarding API to generate a collector configuration and mount any secrets into the collector pod.
+
+### API Specification
+The proposed forwarding API:
+
+```
+apiVersion: "logging.openshift.io/v1alpha1"
+kind: "ClusterLogging"
+spec:
+  forwarding:
```

Comment:
would it make more sense to nest this under the collector since that is where it is most relevant? If a collector isn't defined then this would be moot. (same level as the collector `type`)

Reply:
the feature is "pipeline" so shouldn't "forwarding" instead be "pipeline"  (this struct is defining the pipeline, right?)

Reply:
> the feature is "pipeline" so shouldn't "forwarding" instead be "pipeline" (this struct is defining the pipeline, right?)

I changed it to 'forwarding' because @christianvogt is using that term.

Reply:
> would it make more sense to nest this under the collector since that is where it is most relevant? If a collector isn't defined then this would be moot. (same level as the collector `type`)

I will move it.  The collector is required and I believe we agreed elsewhere that we desired for admins to be explicit

Reply:
Reverted so 'forwarding' is a sibling to collector.  Discussion @sichvoge advised that 'forwarding' may not necessarily be tied to 'collection' in the future